feat(chromium): roll Chromium to r705776

mathiasbynens opened the pull request:
WIP

mathiasbynens committed:
feat(chromium): roll Chromium to r705776

This corresponds to Chromium 79.0.3942.0.

This roll includes:

- Support CSS media feature emulation in CDP
  https://chromium-review.googlesource.com/c/chromium/src/+/1821608
- Implement timezone ID verification
  https://chromium-review.googlesource.com/c/chromium/src/+/1822557
- Allow aria-hidden objects into the browser-side ax tree
  https://chromium-review.googlesource.com/c/chromium/src/+/1760862

mathiasbynens committed:
feat(chromium): roll Chromium to r706915

r706915 corresponds to 79.0.3945.1.

mathiasbynens commented:
```
Failures:
1) Chromium Browser Page Page.screenshot should work for offscreen clip (
screenshot.spec.js:42:5
)
  Message:
expect.toBeGolden failed: screenshot-offscreen-clip.png mismatch! Output is saved in "output-chromium" directory
  Stack:
    Error: expect.toBeGolden failed: screenshot-offscreen-clip.png mismatch! Output is saved in "output-chromium" directory
        at applyMatcher (/tmp/cirrus-ci-build/utils/testrunner/Matchers.js:48:15)
        at /tmp/cirrus-ci-build
        at /tmp/cirrus-ci-build
        at /tmp/cirrus-ci-build/test/screenshot.spec.js:53:26
        at processTicksAndRejections (internal/process/task_queues.js:85:5)
```

@JoelEinbinder @aslushnikov any ideas on what might be going on here?

JoelEinbinder commented on test/accessibility.spec.js:
```diff
@@ -71,6 +71,7 @@ module.exports.addTests = function({testRunner, expect, FFOX}) {
           {role: 'textbox', name: 'Input with whitespace', value: '  '},
           {role: 'textbox', name: '', value: 'value only'},
           {role: 'textbox', name: 'placeholder', value: 'and a value'},
+          {role: 'text', name: 'This is a description!'},
```
This looks like a bug to me

mathiasbynens replied:
The corresponding CL is https://chromium-review.googlesource.com/c/chromium/src/+/1760862

mathiasbynens replied:
For future reference, the relevant CDP return value for this particular node is:

```json
{
  "nodeId": "45",
  "ignored": false,
  "role": {
    "type": "role",
    "value": "text"
  },
  "name": {
    "type": "computedString",
    "value": "This is a description!",
    "sources": [
      {
        "type": "contents",
        "value": {
          "type": "computedString",
          "value": "This is a description!"
        }
      }
    ]
  },
  "properties": [
    {
      "name": "hidden",
      "value": {
        "type": "boolean",
        "value": true
      }
    },
    {
      "name": "hiddenRoot",
      "value": {
        "type": "idref",
        "relatedNodes": [
          {
            "backendDOMNodeId": 11,
            "idref": "desc"
          }
        ]
      }
    }
  ],
  "childIds": [],
  "backendDOMNodeId": 31
}
```

I've amended the logic in `Accessibility.js` to exclude `hidden` nodes from the output for now.

mathiasbynens committed:
Exclude hidden nodes from accessibility tree

mathiasbynens commented:
The `Page.screenshot` offscreen clip failure points to this range: 

> You are probably looking for a change made after 686608 (known good), but no later than 686617 (first known bad).
> https://chromium.googlesource.com/chromium/src/+log/694ff61577da64e0eed8e97eae241e5f484256a9..700c510bbc0ab20b6e2e2ef5684d84d0a885283b

The CL that jumps out is: https://chromium-review.googlesource.com/c/chromium/src/+/1752244 I've left a comment here: https://bugs.chromium.org/p/chromium/issues/detail?id=989251#c21

mathiasbynens committed:
feat(api): page.screenshot now clips to viewport

Per upstream Chromium changes, page.screenshot now clips elements
to the viewport.

This matches the clipping behavior of elements in inner scrollers
(i.e., document and overflow scroll clipping now work the same).

https://bugs.chromium.org/p/chromium/issues/detail?id=937609

mathiasbynens committed:
Fix cross-process navigation per upstream change

https://chromium-review.googlesource.com/c/chromium/src/+/1766001/21#message-a6a61261b97ffca6ecb180c0a2303b538f7a6476

JoelEinbinder commented on test/waittask.spec.js:
```diff
@@ -189,7 +182,8 @@ module.exports.addTests = function({testRunner, expect, product, puppeteer}) {
       await watchdog;
     });
-    it('should survive cross-process navigation', async({page, server}) => {
+    // https://chromium-review.googlesource.com/c/chromium/src/+/1766001/21#message-a6a61261b97ffca6ecb180c0a2303b538f7a6476
```
I didn't look too closely, but I'd guess you need to catch this new error string in the navigation watcher.

mathiasbynens replied:
Thanks for the hint. I think you meant `ExecutionContext`, and I'll add the check there. Let me know if I misunderstood!

JoelEinbinder commented:
Was this supposed to be merged? It just says WIP.

mathiasbynens commented:
> Was this supposed to be merged? It just says WIP.

Yes. The WIP comment is from earlier this week.

For context, this roll was blocking several outstanding PRs + the new v2.0.0 release.

JoelEinbinder commented on test/screenshot.spec.js:
```diff
@@ -39,19 +39,6 @@ module.exports.addTests = function({testRunner, expect, product}) {
       });
       expect(screenshot).toBeGolden('screenshot-clip-rect.png');
     });
-    it('should work for offscreen clip', async({page, server}) => {
```
this looks like a pretty huge change. Are we sure that its ok? At the very least it appears to be a breaking change. It also doesn't appear to match the Firefox behavior now.

mathiasbynens replied:
I learned that there's some history here: https://bugs.chromium.org/p/chromium/issues/detail?id=937609#c4 We'll call it out as a breaking change in the v2 release. The merge commit mentions the following:

> Per upstream Chromium changes, `page.screenshot` now clips elements to the viewport. This matches the clipping behavior of elements in inner scrollers (i.e., document and overflow scroll clipping now work the same).

This seems like a useful change.

JoelEinbinder replied:
If there is new intended behavior, I'd expect to see a new test.

JoelEinbinder replied:
Is it possible to remove the restriction on that crbug? I can't see it.

mathiasbynens replied:
Unfortunately not, but I gave you access before posting my previous comment.

mathiasbynens replied:
> If there is new intended behavior, I'd expect to see a new test.

Sounds reasonable. Let's restore the test, update the expectation, and mark it as failing in Firefox.

mathiasbynens replied:
https://github.com/GoogleChrome/puppeteer/pull/5079
